src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -2116,6 +2116,47 @@ public async Task SendAsync_MultipleExpected100Responses_ReceivesCorrectResponse
             });
         }
 
+        [Fact]
+        public async Task SendAsync_Expect100Continue_RequestBodyFails_ThrowsContentException()
+        {
+            if (IsWinHttpHandler && UseVersion >= HttpVersion20.Value)
+            {
+                return;
+            }
+            if (!TestAsync && UseVersion >= HttpVersion20.Value)
+            {
+                return;
+            }
+
+            var clientFinished = new TaskCompletionSource<bool>();
+            const string ExpectedMessage = "ThrowingContentException";
+
+            await LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpRequestMessage initialMessage = new HttpRequestMessage(HttpMethod.Post, uri) { Version = UseVersion };
+                    initialMessage.Content = new ThrowingContent(() => new Exception(ExpectedMessage));
+                    initialMessage.Headers.ExpectContinue = true;
+                    Exception exception = await Assert.ThrowsAsync<Exception>(() => client.SendAsync(TestAsync, initialMessage));
```

Comment:
Can this be made a more specific exception type?

Reply:
I'm throwing exception of type [`Exception`](https://github.com/dotnet/runtime/blob/bff243c86ca5e0a9467cbb4a8b9bad6aeb90513e/src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs#L2139) from the `ThrowingContent`. I also check for particular message one line lower. Lastly, [`Assert.ThrowsAsync`](https://github.com/xunit/assert.xunit/blob/726e48276895ecba39468990809664948113d876/ExceptionAsserts.cs#L155) matches exactly the exception type, i.e.: if the code throws any derived exception from `Exception`, it will fail.

But I can change it to throw some custom exception if it makes it more readable and less confusing.

Reply:
It can report a false positive if some other piece of code throws the basic `Exception`. I'd also suggest to throw a specific exception type.

Reply:
Changed to a specific exception.